cognite/client/data_classes/data_modeling/instances.py

```diff
@@ -66,7 +66,7 @@ def load(cls, data: dict) -> NodeOrEdgeData:
 
     def dump(self, camel_case: bool = False) -> dict:
         output = asdict(self)
-        if self.source:
+        if self.source and isinstance(self.source, (ContainerId, ViewId)):
             output["source"] = self.source.dump(camel_case)
```

Comment:
When is source not `ContainerId` or `ViewId`?

When doing instance checks like this it's good practice to handle all known types and throw a type error when hitting an unexpected type. Makes debugging issues at runtime much easier.
```suggestion
        if self.source:
		if isinstance(self.source, (ContainerId, ViewId)):
	        	output["source"] = self.source.dump(camel_case)
	        else:
	        	raise TypeError(f"source must be ContainerId or ViewId, was {type(self.source)}")
```

Reply:
The motivation for this was a traceback I got passed, in which source was a dict which is acceptable. But yes will update it to follow this practice.